tests/model_fields/tests.py

```diff
@@ -661,6 +689,15 @@ def test_backend_range_validation(self):
             }
             with self.assertRaisesMessage(ValidationError, expected_message):
                 instance.full_clean()
+            try:
+                with transaction.atomic():
+                    instance.save()
+            except (DataError, IntegrityError):
+                pass  # backends such as Postgres don't fail silently
+            else:
+                qs = self.model.objects.filter(value__lte=min_value)
```

Comment:
The semantics of the test still seem odd to me. I think the expected behavior is that an exception is raised so I would expect to see `assertRaises(...)`. Are any backends besides MySQL expected to reach the "else" block. The test is written in such a way that silent truncation looks like the "normal" case.